config/types/filesystem.go

```diff
@@ -25,15 +25,37 @@ var (
 	ErrFilesystemInvalidFormat = errors.New("invalid filesystem format")
 	ErrFilesystemNoMountPath   = errors.New("filesystem is missing mount or path")
 	ErrFilesystemMountAndPath  = errors.New("filesystem has both mount and path defined")
+	ErrUsedCreateAndMountOpts  = errors.New("cannot use both create object and mount-level options fields")
+	ErrWarningCreateDeprecated = errors.New("the create object has been deprecated in favor of mount-level options")
 )
 
 func (f Filesystem) Validate() report.Report {
 	r := report.Report{}
 	if f.Mount == nil && f.Path == nil {
-		return report.ReportFromError(ErrFilesystemNoMountPath, report.EntryError)
+		r.Add(report.Entry{
+			Message: ErrFilesystemNoMountPath.Error(),
+			Kind:    report.EntryError,
+		})
 	}
-	if f.Mount != nil && f.Path != nil {
-		return report.ReportFromError(ErrFilesystemMountAndPath, report.EntryError)
+	if f.Mount != nil {
+		if f.Path != nil {
+			r.Add(report.Entry{
+				Message: ErrFilesystemMountAndPath.Error(),
+				Kind:    report.EntryError,
+			})
+		}
+		if len(f.Mount.Options) > 0 && f.Mount.Create != nil {
```

Comment:
Can you put this block inside of the lower one? I think it might be a bit easier to follow.